### PR TITLE
OCM-10952 | test: Automated ids:45502,46308,45509,46310 for proxy supporting

### PR DESCRIPTION
**Pull request by xueli181114**

### CODE REVIEW

File: tests/e2e/test_rosacli_cluster.go [hunk 1]
```diff
@@ -1381,6 +1620,120 @@ var _ = Describe("Create cluster with invalid options will",
...
+					"--http-proxy", "http://example.com",
+					"--https-proxy", "https://example.com",
+				)
+				Expect(err).To(HaveOccurred())
+				Expect(output.String()).Should(ContainSubstring("The number of subnets for a"))
+
+				By("Prepare vpc with subnets")
+				vpc, err := vpc_client.PrepareVPC(clusterName, constants.CommonAWSRegion, "", true, "")
+				Expect(err).ToNot(HaveOccurred())
+				// defer vpc.DeleteVPCChain(true)
```
Comment (86254860) [inline tests/e2e/test_rosacli_cluster.go:1639]: Why don't delete VPC and subnets resources at the end?

File: tests/e2e/test_rosacli_cluster_post.go [hunk 2]
```diff
@@ -536,6 +536,46 @@ var _ = Describe("Healthy check",
...
+				Expect(output.String()).Should(
+					MatchRegexp(`-{2}no-proxy strings[\s\t]+A comma-separated list of destination domain names`))
+				Expect(output.String()).Should(
+					MatchRegexp(`-{2}additional-trust-bundle-file string[\s\t]+A file`))
+
+				By("Check the proxy is set correctly")
+				clusterDescription, err := clusterService.DescribeClusterAndReflect(clusterID)
+				Expect(err).ToNot(HaveOccurred())
+				if profile.ClusterConfig.ProxyEnabled {
+					clusterHTTPProxy, clusterHTTPSProxy, clusterNoProxy := "", "", ""
```
Comment (86254860) [inline tests/e2e/test_rosacli_cluster_post.go:558]: There're two same code blocks. Could we abstract them to a common file?

File: tests/e2e/test_rosacli_cluster.go [hunk 3]
```diff
@@ -490,8 +383,353 @@ var _ = Describe("Edit cluster",
...
+					"--additional-trust-bundle-file", originalCAFile,
+				)
+				Expect(err).ToNot(HaveOccurred())
+				verifyProxy(updateHttpProxy, updateHttpsProxy, updatedNoProxy, updatedCA)
+
+				By("Edit cluster for removing cluster-wide proxy")
+				updateHttpProxy = ""
+				updateHttpsProxy = ""
+				updatedNoProxy = ""
+				updatedCA = originalCAFile
```
[the remark this answers is not in the document]
Comment (xueli181114) [reply tests/e2e/test_rosacli_cluster.go:455]: Check line 434, it had set it to "". And this is another scenario which will update the CA back.

File: tests/e2e/test_rosacli_cluster.go (hunk 1 above)
Comment (xueli181114) [reply tests/e2e/test_rosacli_cluster.go:1639]: Oh. forget to comment it back. Updating...

Comment (xueli181114) [reply tests/e2e/test_rosacli_cluster.go]: Uncommented

**Pull request merged**